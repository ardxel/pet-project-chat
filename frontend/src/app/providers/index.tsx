import { Suspense } from 'react';
import { ErrorBoundary } from 'react-error-boundary';
import { Outlet } from 'react-router';
import { BaseErrorPage } from 'shared/ui';
import { Header } from 'widgets';
import StoreProvider from './store.provider';

const Layout = () => (
  <>
    <ErrorBoundary fallback={<BaseErrorPage />}>
      <StoreProvider>
        <Suspense fallback={<h1 className='text-red text-3xl'>Loading...</h1>}>
          <Header />
          <Outlet />
        </Suspense>
      </StoreProvider>
    </ErrorBoundary>
  </>
);

export default Layout;
