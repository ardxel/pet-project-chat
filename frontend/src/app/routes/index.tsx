import { lazy } from 'react';
import { Navigate, Outlet, createBrowserRouter } from 'react-router-dom';
import { Paths } from 'shared/routing';
import { Header } from 'widgets';

const LoginPage = lazy(() => import('pages/login'));
const RegistrationPage = lazy(() => import('pages/register'));
const ChatPage = lazy(() => import('pages/chat'));

const router = createBrowserRouter([
  {
    path: Paths.home,
    element: (
      <>
        <Header />
        <Outlet />
      </>
    ),
    children: [
      {
        path: Paths.chat,
        element: <ChatPage />,
      },
      {
        path: Paths.contacts,
        element: <div></div>,
      },
      {
        path: Paths.stories,
        element: <div></div>,
      },
      {
        path: Paths.band,
        element: <Navigate to={Paths.home} />,
      },
    ],
  },
  {
    path: Paths.home,
    children: [
      {
        path: Paths.login,
        element: <LoginPage />,
      },
      {
        path: Paths.registration,
        element: <RegistrationPage />,
      },
      {
        path: Paths.band,
        element: <Navigate to={Paths.home} />,
      },
    ],
  },
]);

export default router;
