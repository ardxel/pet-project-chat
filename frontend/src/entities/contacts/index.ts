export * from './api/contacts.api';
export * from './api/types';
export * from './model';
