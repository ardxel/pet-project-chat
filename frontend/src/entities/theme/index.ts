export * from './lib/Theme.provider';
export * from './model';
