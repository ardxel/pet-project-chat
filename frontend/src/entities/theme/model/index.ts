export * from './theme.slice';
export * from './types';
