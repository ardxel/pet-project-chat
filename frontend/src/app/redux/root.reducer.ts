import { combineReducers } from '@reduxjs/toolkit';
import { contactsSlice } from 'entities/contacts';
import { sessionSlice } from 'entities/session';
import { themeSlice } from 'entities/theme';
import { baseApi } from 'shared/api/baseApi';

export const rootReducer = combineReducers({
  [contactsSlice.name]: contactsSlice.reducer,
  [sessionSlice.name]: sessionSlice.reducer,
  [themeSlice.name]: themeSlice.reducer,
  [baseApi.reducerPath]: baseApi.reducer,
});
