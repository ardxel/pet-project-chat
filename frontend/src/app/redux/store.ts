import { configureStore } from '@reduxjs/toolkit';
import { sessionSlice } from 'entities/session';
import { themeSlice } from 'entities/theme';
import { invalidateAccessTokenListener } from 'features/auth/invalidateAccessToken';
import { FLUSH, PAUSE, PERSIST, PURGE, REGISTER, REHYDRATE, persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import { baseApi, config } from 'shared/api';
import { rootReducer } from './root.reducer';

const persistConfig = {
  key: 'root',
  storage,
  whiteList: [sessionSlice.name, themeSlice.name],
};

export const store = configureStore({
  devTools: config.isDev,
  reducer: persistReducer(persistConfig, rootReducer),
  middleware: (getDefailtMiddleware) =>
    getDefailtMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }).concat(baseApi.middleware, invalidateAccessTokenListener.middleware),
});

export const persistedStore = persistStore(store);
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof rootReducer>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
