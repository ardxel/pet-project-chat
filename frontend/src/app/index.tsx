import { persistedStore, store } from 'app/redux';
import { ThemeProvider } from 'entities/theme';
import { createRoot } from 'react-dom/client';
import { ErrorBoundary } from 'react-error-boundary';
import { Provider as ReduxProvider } from 'react-redux';
import { RouterProvider } from 'react-router';
import { PersistGate } from 'redux-persist/integration/react';
import { BaseErrorPage } from 'shared/ui';
import router from './routes';
import './styles.css';

function bootstrap() {
  return createRoot(document.getElementById('root') as HTMLDivElement).render(
    <>
      <ErrorBoundary fallback={<BaseErrorPage />}>
        <ReduxProvider store={store}>
          <PersistGate loading={null} persistor={persistedStore}>
            <ThemeProvider>
              <RouterProvider router={router} />
            </ThemeProvider>
          </PersistGate>
        </ReduxProvider>
      </ErrorBoundary>
    </>,
  );
}

bootstrap();
