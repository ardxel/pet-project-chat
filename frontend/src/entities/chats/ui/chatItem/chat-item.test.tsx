import { store } from 'app/redux';
import { ThemeProvider } from 'entities/theme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';
import { ChatListItem } from '.';

const userStub = { name: 'John Doe' };

describe('Test ChatListItem', () => {
  describe('Test ChatListItem by snapshots', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <ThemeProvider>
            <ChatListItem user={userStub} conversationId='none' />
          </ThemeProvider>
        </Provider>
      </MemoryRouter>,
    );

    const tree = component.toJSON();
    test('create or compare snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
