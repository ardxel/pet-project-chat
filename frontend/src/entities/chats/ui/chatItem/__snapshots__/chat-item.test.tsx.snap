// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ChatListItem Test ChatListItem by snapshots create or compare snapshot 1`] = `
<div
  className="flex items-center justify-between cursor-pointer hover:bg-gray-100 dark:hover:bg-gray-900 px-4 py-3 relative"
  data-testid="ddd"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="flex gap-x-3"
  >
    <div
      className="relative w-[50px] h-[50px]"
    >
      <div
        className="relative border border-border w-full h-full rounded-md"
      >
        <h1
          className="text-center text-2xl leading-none  absolute left-1/2 -transform translate-y-1/2 -translate-x-1/2"
        >
          J
        </h1>
      </div>
    </div>
    <div>
      <div
        className="flex items-center h-full"
      >
        <h4
          className="text-left text-sm"
        >
          John Doe
        </h4>
      </div>
    </div>
  </div>
  <div />
</div>
`;
